lib/galaxy/model/store/ro_crate_utils.py

```diff
@@ -220,7 +220,77 @@ def _add_workflows(self, crate: ROCrate):
 
             crate.license = self.workflow.license or ""
             crate.mainEntity["name"] = self.workflow.name
-            crate.mainEntity["subjectOf"] = cwl_wf
+
+            # Adding the creator information
+            if hasattr(self.workflow, 'creator_metadata') and self.workflow.creator_metadata:
+                creators = self.workflow.creator_metadata
+                if creators and isinstance(creators, list) and len(creators) > 0:
+                    first_creator = creators[0]
+                    creator_entity = crate.add(
+                        ContextEntity(
+                            crate,
+                            first_creator.get('identifier', ''),  # Default to empty string if identifier is missing
+                            properties={
+                                "@type": "Person",
+                                "name": first_creator.get('name', ''),  # Default to empty string if name is missing
+                                "orcid": first_creator.get('identifier', ''),  # Assuming identifier as orcid, or adjust accordingly
+                            },
+                        )
+                    )
+                    crate.mainEntity.append_to("creator", creator_entity)
+
+            # Add CWL workflow entity if exists
+            crate.mainEntity["subjectOf"] = cwl_wf if cwl_wf else ""
+
+        # Add tools used in the workflow
+        self._add_tools(crate)
+
+    def _add_tools(self, crate: ROCrate):
+        tool_entities = []
+
+        # Iterate over each step in the workflow
+        for step in self.workflow.steps:
+            # Check if the step corresponds to a tool
+            if step.type == "tool":
+                tool_id = step.tool_id
```

Comment:
This will list tools as many times as they appear in a workflow, is that the intention ?

Reply:
For now with the info we put in the tools properties maybe not but I don't know maybe one day with another update it could be interesting to have the inputs and outputs in the tools properties. Do you think I should keep only one occurence of the tool and not every time it appears in the workflow ?

Reply:
Is that what `@SoftwareApplication` is about ? Because those parameters are encoded in the workflow itself. I would assume this is just a high-level summary of software used ?

Reply:
It should be a high level summary lifted from the workflow specification into RO-Crate - see mention of `SoftwareApplication` in https://www.researchobject.org/workflow-run-crate/profiles/provenance_run_crate/

Ideally then connected to each of the output of each tool invocation, but it's OK if the listing of tools are just in the prospective part (blue bit of diagram)

Reply:
You can use `HowToStep` to indicate each placement of the tool in the workflow, then they should refer to the same `SoftwareApplication` (assuming it's the same version etc)

Reply:
Oh okay I'll try to manage that then ! Thanks a lot for your input @stain !!